### PR TITLE
feat: enable composer to call a remote skill 

### LINKED ISSUE #696: Call a remote skill
## Task Track

Open Item
- 8/30 Depends on SDK

https://microsoft-my.sharepoint.com/:w:/r/personal/luhan_microsoft_com/_layouts/15/Doc.aspx?sourcedoc=%7B03AFB829-07F7-4764-823A-842531D987DC%7D&file=RemoteSkillFuncSpec.docx&action=default&mobileredirect=true

see microsoft/botbuilder-dotnet#2324

### PR DESCRIPTION
**Pull request by luhan2017**
## Description

In this PR, I updated the bot project and enable composer to connect to a remote skill. 
Note: A remote skill right now could not be a normal skill, It should have the ability to send and handle EndOfConversation type of Activity.  Refer to https://github.com/microsoft/botbuilder-dotnet/tree/master/FunctionalTests/Skills/SimpleBotToBot/EchoSkillBot to create a remote skill.

How to connect to a remote skill:

Step1: configure settings. to call a remote skill, the bot must be registered with a MicrosoftAppID and MicrosoftPassword, and also SkillHostEndpoint, which is the skill endpoint of the current bot. 

  "MicrosoftAppPassword": "",
  "MicrosoftAppId": "",
  "SkillHostEndpoint": "http://localhost:3978/api/skills/" // http://localhost:3978 is the endpoint of the caller not callee. 

Step2: add SkillDialog in the bot, for example:
{
  "$kind": "Microsoft.SkillDialog",
  "skillAppId": "3083d5b8-a3b6-4d50-82c7-af10aa96b20c",   // the remote skill MicrosoftAppId
  "skillEndpoint": "http://localhost:1205/api/messages",  // the remote skill endpoint
  "activity": "Hi"   
}

Note: step2 can be added in Composer UX after the UX part is ready. 

Step3: Click StartBot and Test in Emulator.

This PR is only to enable Composer to call a remote skill. It doesn't include creating a "remote skill" using composer. 

## Task Item

fixes #696 

## Screenshots

image

### CODE REVIEW

**Comment by cwhitten** [comment]
@luhan2017 can we introduce the AdaptiveSkillDialog component schema in this PR as well?

File: Composer/packages/extensions/visual-designer/src/schema/uischema.tsx [hunk 1]
```diff
@@ -152,6 +152,27 @@ export const uiSchema: UISchema = {
...
+    },
+    body: {
+      'ui:widget': DialogRef,
+      dialog: data => data.dialog,
+      getRefContent: data => dialogRef => (
+        <>
+          {dialogRef || '?'} <FixedInfo>(Dialog)</FixedInfo>
+        </>
+      ),
+    },
```
Comment (yeze322) [inline Composer/packages/extensions/visual-designer/src/schema/uischema.tsx]: @yeze322 update the content displayed here

Comment (cwhitten) [reply Composer/packages/extensions/visual-designer/src/schema/uischema.tsx]: Ah, yeah I haven't cleaned up the node renderer yet!

Comment (yeze322) [reply Composer/packages/extensions/visual-designer/src/schema/uischema.tsx]: It's updated

**Pull request merged**